sympy/solvers/solvers.py

```diff
@@ -1337,6 +1337,9 @@ def _solve(f, *symbols, **flags):
     if f.is_Mul:
         result = set()
         for m in f.args:
+            if m in set([-oo, zoo, oo]):
```

Comment:
I would suggest writing this as `[S.NegativeInfinity, S.Infinity, S.ComplexInfinity]`.